kernel/net/net_tcp.c

```diff
@@ -1808,6 +1809,48 @@ static int net_tcp_setsockopt(net_socket_t *hnd, int level, int option_name,
                 case SO_ERROR:
                 case SO_TYPE:
                     goto ret_inval;
+
+                case SO_RCVBUF:
+
+                    if(option_len != sizeof(uint32_t))
+                        goto ret_inval;
+
+                    tmp = sock->rcvbuf_sz;
+
+                    /* Maximum size for TCP window is currently 65535 */
+                    sock->rcvbuf_sz = (*(uint32_t *)option_value > 65535) ?
+                                      65535 : *(uint32_t *)option_value;
+
+                    new_ptr = realloc(sock->data.rcvbuf, sock->rcvbuf_sz);
+                    if(!new_ptr) {
+                        sock->rcvbuf_sz = tmp;
+                        goto ret_nomem;
+                    }
+
+                    sock->data.rcvbuf = new_ptr;
+
+                    goto ret_success;
+
+                case SO_SNDBUF:
+
+                    if(option_len != sizeof(uint32_t))
+                        goto ret_inval;
+
+                    tmp = sock->sndbuf_sz;
+
+                    /* Maximum size for TCP window is currently 65535 */
+                    sock->sndbuf_sz = (*(uint32_t *)option_value > 65535) ?
+                                      65535 : *(uint32_t *)option_value;
+
+                    new_ptr = realloc(sock->data.sndbuf, sock->sndbuf_sz);
+                    if(!new_ptr) {
+                        sock->sndbuf_sz = tmp;
+                        goto ret_nomem;
+                    }
+
+                    sock->data.sndbuf = new_ptr;
```

Comment:
The same suggestion from up above with the receive buffer applies here too. The minimum Linux specifies for this value is 2048.